trestle/transforms/utils/osco_helper.py

```diff
@@ -0,0 +1,444 @@
+        inventory.props = props
+        inventory.implemented_components = [ImplementedComponent(component_uuid=self._get_component_ref(rule_use))]
+        self.inventory_map[key] = inventory
+
+    def _get_inventory_ref(self, rule_use: RuleUse) -> t_inventory_ref:
+        """Get inventory reference for specified RuleUse."""
+        key = rule_use.target + ':' + rule_use.name
+        return self.inventory_map[key].uuid
+
+    def _observation_extract(self, rule_use: RuleUse) -> None:
+        """Extract observation from RuleUse."""
+        observation = Observation(
+            uuid=str(uuid.uuid4()), description=rule_use.idref, methods=['TEST-AUTOMATED'], collected=self.timestamp
+        )
+        subject_reference = SubjectReference(uuid_ref=self._get_inventory_ref(rule_use), type='inventory-item')
+        observation.subjects = [subject_reference]
+        props = []
+        props.append(Property(name='scanner_name', value=rule_use.scanner_name, ns=self.ns))
+        props.append(Property(name='scanner_version', value=rule_use.scanner_version, ns=self.ns))
+        props.append(Property(name='idref', value=rule_use.idref, ns=self.ns, class_='scc_goal_description'))
+        props.append(Property(name='idref', value=rule_use.idref, ns=self.ns, class_='scc_check_id_name'))
+        props.append(Property(name='version', value=rule_use.version, ns=self.ns, class_='scc_check_version'))
+        props.append(Property(name='result', value=rule_use.result, ns=self.ns, class_='scc_result'))
+        props.append(Property(name='time', value=rule_use.time, ns=self.ns, class_='scc_timestamp'))
+        props.append(Property(name='severity', value=rule_use.severity, ns=self.ns, class_='scc_check_severity'))
+        props.append(Property(name='weight', value=rule_use.weight, ns=self.ns))
+        props.append(Property(name='benchmark_href', value=rule_use.benchmark_href, ns=self.ns))
+        props.append(
+            Property(name='benchmark_id', value=rule_use.benchmark_id, ns=self.ns, class_='scc_predefined_profile')
+        )
+        observation.props = props
+        self.observation_list.append(observation)
+        rule_use.observation = observation
+
+    def _finding_extract(self, rule_use: RuleUse) -> None:
+        """Extract finding from RuleUse."""
+        if rule_use.result not in self.results_map.keys():
+            self.results_map[rule_use.result] = 0
+        self.results_map[rule_use.result] += 1
+
+    def _process(self, co_report: ComplianceOperatorReport) -> None:
+        """Process ingested data."""
+        rule_use_generator = co_report.rule_use_generator()
+        try:
+            while True:
+                rule_use = next(rule_use_generator)
+                self._component_extract(rule_use)
+                self._inventory_extract(rule_use)
+                self._observation_extract(rule_use)
+                self._finding_extract(rule_use)
```

Comment:
As a generator can't you iterate over this using `for rule_use in rule_use_generator` which would avoid the awkward try except block?

Reply:
Yep, improved. Thx!